onnxruntime/core/framework/finalize_session_state.cc

```diff
@@ -82,6 +82,23 @@ Status FinalizeSessionState(SessionState& session_state,
       ITensorAllocator::Create(session_state.GetEnableMemoryPattern(), *exec_plan_ptr, session_state,
                                session_state.GetMutableWeightsBuffers()));
 
+#ifdef ENABLE_TRAINING
+  if (session_state.GetEnableMemoryPattern()) {
```

Comment:
consider also adding "&& logging at INFO is enabled"

Reply:
Fair